File: data/EXP01/python/2406/2406.output_3.py
# @lc app=leetcode id=2406 lang=python3
class Solution:
    def minGroups(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[0])  # Sort intervals based on start time
        groups = 0
        start = end = intervals[0][0]
        
        for interval in intervals:
            if interval[0] > end:
                start = interval[0]
                end = interval[1]
                groups += 1
            else:
                end = max(end, interval[1])
        
        return groups + 1  # Add 1 to include the last group

File: data/EXP01/golang_no_system_var/2406/2406.output_3.go
func minGroups(intervals [][]int) int {
    sort.Slice(intervals, func(i, j int) bool {
        return intervals[i][0] < intervals[j][0]
    })
    
    groups := 1
    prevEnd := intervals[0][1]
    
    for i := 1; i < len(intervals); i++ {
        if intervals[i][0] > prevEnd {
            groups++
        }
        prevEnd = max(prevEnd, intervals[i][1])
    }
    
    return groups
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}
